### ISSUE TITLE
ethminer 0.18.0-rc.0 EthereumStratum protocol broken again

### ISSUE THREAD

**Issue by Rom1kz** [ethereum-mining/ethminer#1866]
**Environment**
- Operating System: Ubuntu
- Hardware 1060 6G
- Ethminer 0.18.0-rc.0
- Pool: comining.io

**Start line**
ethminer.exe -U -P stratum2+tcp://AGQZgM6jtuYmVxNenWabGF5.worker@s-ru.comining.io:9999 --work-timeout 1000

Here is the pool debug output
```
stratum data in '{"id":1,"method":"mining.subscribe","params":["ethminer-0.18.0-rc.0","EthereumStratum/1.0.0"]}'
subscribed
allocated extranonce 'c502'
stratum data out: {"id":1,"result":[["mining.notify","8e38e0a3df0d461094df38e56ed869bb","EthereumStratum/1.0.0"],"c502"],"error":null}
stratum data in '{"id":2,"method":"mining.extranonce.subscribe","params":[]}'
stratum data out: {"id":2,"result":true,"error":null}
stratum data in '{"id":3,"method":"mining.authorize","params":["AGQZgM6jtuYmVxNenWabGF5.worker","X"]}'
stratum data out: {"id":3,"result":true,"error":null}
```

and some time later i get
```
stratum data in '{"id":47,"method":"mining.submit","params":["AGQZgM6jtuYmVxNenWabGF5.worker","2e497874523144fd8e08f9d1260772c9","3f041de8dfb5432d"]}'
stratum data out: {"id":47,"result":false,"error":{"code":-1,"message":"rejected"}}

stratum data in '{"id":40,"method":"mining.submit","params":["AGQZgM6jtuYmVxNenWabGF5.worker","20f70aa73c054871ad484ee38a839624","3f041de1fecd772c"]}'
stratum data out: {"id":40,"result":false,"error":{"code":-1,"message":"rejected"}}
```

miner sent solutions with static part 3f041de... looks like it use 3f041de as extranonce
and yes - miner sent entire nonce, not part

**Comment by lesjokolat** [comment]
what happens if you try this(-v 3 is more verbose messaging):

ethminer.exe -U -P stratum1+tcp://AGQZgM6jtuYmVxNenWabGF5.worker@s-ru.comining.io:9999 --work-timeout 1000 -v 3

**Comment by Rom1kz** [comment]
```
 i 23:59:14 ethminer Configured pool 172.24.24.24:9999
 i 23:59:14 ethminer Selected pool 172.24.24.24:9999
 i 23:59:14 ethminer  >> {"id":1,"method":"mining.subscribe","params":["ethminer-0.18.0-rc.0","EthereumStratum/1.0.0"]}
 i 23:59:14 ethminer  << {"id":1,"result":[["mining.notify","8ade8390e96841bca7ba53dcdad5c25e","EthereumStratum/1.0.0"],"d336"],"error":null}
 i 23:59:14 ethminer Stratum mode : EthereumStratum/1.0.0 (NiceHash)
 i 23:59:14 ethminer Established connection to 172.24.24.24:9999
 i 23:59:14 ethminer Spinning up miners...
cu 23:59:14 cuda-0   Using Pci Id : 01:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
cu 23:59:14 cuda-1   Using Pci Id : 02:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
cu 23:59:14 cuda-2   Using Pci Id : 03:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
cu 23:59:14 cuda-3   Using Pci Id : 04:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
cu 23:59:14 cuda-4   Using Pci Id : 05:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
cu 23:59:14 cuda-5   Using Pci Id : 06:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
cu 23:59:14 cuda-6   Using Pci Id : 09:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
cu 23:59:14 cuda-7   Using Pci Id : 0a:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
cu 23:59:14 cuda-8   Using Pci Id : 0b:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
cu 23:59:14 cuda-9   Using Pci Id : 0c:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
cu 23:59:14 cuda-10  Using Pci Id : 0e:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
cu 23:59:14 cuda-11  Using Pci Id : 0f:00.0 GeForce GTX 1060 6GB (Compute 6.1) Memory : 5.93 GB
 i 23:59:14 ethminer  >> {"id":2,"method":"mining.extranonce.subscribe","params":[]}
 i 23:59:14 ethminer  >> {"id":3,"method":"mining.authorize","params":["AGQZgM6jtuYmVxNenWabGF5.worker","X"]}
 i 23:59:14 ethminer  << {"id":2,"result":true,"error":null}
 i 23:59:15 ethminer  << {"id":3,"result":true,"error":null}
 i 23:59:15 ethminer Authorized worker AGQZgM6jtuYmVxNenWabGF5.worker
```

```
 i 00:00:19 ethminer  >> {"id":49,"method":"mining.submit","params":["AGQZgM6jtuYmVxNenWabGF5.worker","03fb89ab06374a16bceeb894c978149c","1268948a99724cf7"]}
 i 00:00:19 ethminer  << {"id":49,"result":false,"error":{"code":-1,"message":"rejected"}}
 X 00:00:19 ethminer Reject reason : code:-1 message:rejected
 X 00:00:19 ethminer **Rejected  33 ms. 172.24.24.24:9999

```

**Comment by lesjokolat** [comment]
ok try this:

ethminer.exe -U -P stratum1+tcp://AGQZgM6jtuYmVxNenWabGF5%%2eworker@s-ru.comining.io:9999 --work-timeout 1000 -v 3

**Comment by Rom1kz** [comment]
only ethereum stratum proto is allowed on comining.io
so stratum1+tcp wont work

**Comment by Rom1kz** [comment]
but i've enabled ethproxy on my account for testing
and miner works fine

```
 i 10:16:16 ethminer  >> {"id":46,"method":"eth_submitWork","params":["0x0d53c16e47bb150c","0xe9f5b8b9629afc667755f04d6164bf4e798bddef8cd84308886cfa0613ab6746","0x30426dacd51035fa6da022435c1df18a2ba7a28817183146417455a878889abe"],"worker":"worker"}
 i 10:16:17 ethminer  << {"id":46,"result":true,"error":null}
 i 10:16:17 ethminer **Accepted 68 ms. s-ru.comining.io [46.17.42.144:9999]

```

**Comment by bmatthewshea** [comment]
So, you are using a nicehash proxy?  
Because that is not a nicehash address.  

If not a nicehash connection, try `stratum` or `stratum1` as mentioned.

From `--help`:

```
stratum variants:
      stratum -   Stratum
      stratum1 -  Eth proxy
      stratum2 -  Eth stratum (nicehash)
```

**Comment by Rom1kz** [comment]
i use comining.io pool which supports nicehash protocol

**Comment by bmatthewshea** [comment]
> i use comining.io pool which supports nicehash protocol

Why use a 3rd party pool for nicehash?
I guess you want even more people in the middle taking fee's?
Nicehash is already a "pooled" connection.

Just connect directly using ethminer.exe (or a proxy that supports nicehash stratum).

The algo you will use is of course DaggarHashimoto:

![SHEA99-2019-08-11_144944](https://user-images.githubusercontent.com/984097/62838869-8c0ef080-bc47-11e9-9127-3cb73a5fb6ca.png)

Using the above image, we can see our connection string which we can use in this basic command:

`ethminer.exe" --opencl -P stratum2+tcp://wallet.worker@daggerhashimoto.usa.nicehash.com:3353`

**Comment by AndreaLanfranchi** [comment]
Solved by #1983
